snap/local/hooks/cmd/configure/configure.go

```diff
@@ -20,26 +20,92 @@ package main
 
 import (
 	"fmt"
+	"io/ioutil"
 	"os"
+	"path/filepath"
+	"strings"
 
 	hooks "github.com/canonical/edgex-snap-hooks/v2"
 )
 
 var cli *hooks.CtlCli = hooks.NewSnapCtl()
 
 const ( // iota is reset to 0
-	devVirtService = iota
-	kuiperService
+	kuiperService = iota
 	secProxyService
 	secStoreService
 	otherService
 )
 
 const (
-	ON  = "on"
-	OFF = "off"
+	ON      = "on"
+	OFF     = "off"
+	UNSET   = ""
 )
 
+// getKuiperServices returns the list of servics used for
+// EdgeX rules-engine processing.
+func getKuiperServices() []string {
+	return []string{hooks.ServiceAppCfg, hooks.ServiceKuiper}
+}
+
+// getProxyServices returns the list of services which implement
+// the API Gateway. Note this list *excludes* Consul and the
+// Secret Store services.
+func getProxyServices() []string {
+	return []string{"postgres", "kong-daemon", "security-proxy-setup"}
+}
+
+// getSecretStoreServices returns the list of services which implement
+// the Secret Store and related dependencies (i.e. the services that
+// secure redis and consul which are tightly bound to the secret store
+// being enabled).
+func getSecretStoreServices() []string {
+	return []string{"security-secretstore-setup", "vault",
+		"security-consul-bootstrapper", "security-bootstrapper-redis"}
+}
+
+// getEdgeXServices returns the list of EdgeX services in the snap
+func getEdgeXServices() []string {
+	return []string{"core-data", "core-metadata", "core-command",
+			"device-virtual", "support-notifications",
+			"support-scheduler", "sys-mgmt-agent"}
+}
+
+// getRequiredServices returns the minimum list of required
+// snap services for a working EdgeX instance.
+func getRequiredServices() []string {
+	return []string{"consul", "redis", "core-metadata"}
+}
+
+// getCoreDefaultServices returns the list of core services
+// that are started by default (in addition to the required
+// services)
+func getCoreDefaultServices() []string {
+	return []string{"core-command", "core-data"}
+}
```

Comment:
Just my personal preference, feel free to skip:
- Use global variables instead of getter functions. This is a "main" and small package, so accidental modifications to those slices from some function are not likely.
- Add line breaks between each line. This will improve readability and more importantly allow changes to different parts from independent PRs.

Reply:
Line breaks are another thing we should discuss.  I've always used blank lines freely in C/C++/Java/... Over time, I've been adding less and less of them to my Go code, as that most Go developers seems to follow this practice.